src/Analyzers/Core/CodeFixes/RemoveRedundantEqualityWithTrue/RemoveRedundantEqualityWithTrueCodeFixProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEqualityWithTrue
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+
+    internal sealed class RemoveRedundantEqualityWithTrueCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public RemoveRedundantEqualityWithTrueCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.RemoveRedundantEqualityWithTrueDiagnosticId);
+
+        internal override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    AnalyzersResources.Remove_redundant_equality_with_true,
+                    (ct) => FixAsync(context.Document, diagnostic, ct)),
+                    diagnostic);
+            }
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.Location.SourceSpan);
```

Comment:
```suggestion
                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnertmostNodeForTie: true);
```

note: withoutthis change, your fix is buggy and will crash in the following case:  `Foo(x == true)`, please add test.